src/multiscale/multiscale.jl

```diff
@@ -1,33 +1,131 @@
-function multiscale(alg::MultiScaleAlgorithm, e::ComplexityEstimator, x::AbstractVector;
-        maxscale::Int = 8)
+# This file contains an API for multiscale (coarse-grained/downsampled) computations.
+# This API is not final nor agreed upon yet. Nothing is exported or documented
+# as part of public API. It must be considered as work in progress.
+export downsample
+export multiscale, multiscale_normalized
+export MultiScaleAlgorithm
 
-    downscaled_timeseries = [downsample(alg, s, x) for s in 1:maxscale]
-    return complexity.(Ref(e), downscaled_timeseries)
-end
+using Statistics
+
+"""
+    MultiScaleAlgorithm
+
+The supertype for all multiscale coarse-graining/downsampling algorithms. Concrete subtypes
+are:
+
+- [`RegularDownsampling`](@ref)
+- [`CompositeDownsampling`](@ref)
+"""
+abstract type MultiScaleAlgorithm end
+
+"""
+    downsample(algorithm::MultiScaleAlgorithm, s::Int, x)
+
+Downsample and coarse-grain `x` to scale `s` according to the given
+[`MultiScaleAlgorithm`](@ref). The return type depends on `algorithm`.
+"""
+downsample(method::MultiScaleAlgorithm, s::Int, x)
+
+downsample(alg::MultiScaleAlgorithm,  s::Int, x::AbstractStateSpaceSet) =
+    StateSpaceSet(map(t -> downsample(alg, s, t)), columns(x)...)
+
+"""
+    multiscale(algorithm::MultiScaleAlgorithm, [args...], x; maxscale::Int = 8)
+
+A convenience function to compute the multiscale version of any
+[`InformationMeasureEstimator`](@ref) or [`ComplexityEstimator`](@ref).
+
+## Description
+
+This function generalizes of the [Costa2002](@cite) multiscale entropy to any discrete
+information measure, any differential information measure, and any other complexity measure.
+
+Utilizes [`downsample`](@ref) with the given `algorithm` to first produce coarse-grained,
+downsampled versions of `x` for scale factors `1:maxscale`. Then, [`information`](@ref) or
+[`complexity`](@ref), depending on the input arguments, is applied to each of
+the coarse-grained timeseries. If `N = length(x)`, then the length of the most severely
+downsampled version of `x` is `N ÷ maxscale`, while for scale factor `1`, the original
+time series is considered.
 
-function multiscale_normalized(alg::Regular, e::ComplexityEstimator, x::AbstractVector;
-        maxscale::Int = 8)
-    downscaled_timeseries = [downsample(alg, s, x) for s in 1:maxscale]
-    return complexity_normalized.(Ref(e), downscaled_timeseries)
+## Coarse-graining algorithms
+
+The available downsampling routines are:
+
+- [`RegularDownsampling`](@ref) yields a single `Vector` per scale.
+- [`CompositeDownsampling`](@ref) yields a `Vector{Vector}` per scale.
+
+## Keyword Arguments
+
+- `maxscale::Int`. The maximum number of scales (i.e. levels of downsampling). The actual
+    maximum scale level is `length(x) ÷ 2`, but to avoid applying the method to time
+    series that are extremely short, maybe consider limiting `maxscale` (e.g.
+    `maxscale = length(x) ÷ 5`).
```

Comment:
I think it would be better to allow users to provide **specific scales** i.e., `AbstractVector{Int}` instead. If users provide an `Int`, then `1:int` is taken as the scales. This is more flexible and it doesn't costs us anything in code complexity.